Jenkinsfile

```diff
@@ -20,35 +20,96 @@ pipeline {
+                                checkout scm
+                                withMaven(maven: 'M3', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice-maven-settings') {
+                                    withCredentials([string(credentialsId: 'ddf-coverity-token', variable: 'COVERITY_TOKEN')]) {
+                                        withEnv(["PATH=${tool 'coverity-linux'}/bin:${env.PATH}"]) {
+                                            configFileProvider([configFile(fileId: 'coverity-maven-settings', replaceTokens: true, variable: 'MAVEN_SETTINGS')]) {
+                                                echo sh(returnStdout: true, script: 'env')
+                                                sh 'cov-build --dir cov-int mvn -DskipTests=true -Dfindbugs.skip=true -Dcheckstyle.skip=true -Djacoco.skip=true install -pl !distribution/docs --settings $MAVEN_SETTINGS'
+                                                sh 'tar czvf ddf.tgz cov-int'
+                                                sh 'curl --form token=$COVERITY_TOKEN --form email=cmp-security-team@connexta.com --form file=@ddf.tgz --form version="master" --form description="Description: DDF CI Build" https://scan.coverity.com/builds?project=codice%2Fddf'
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        },
+                        nodesecurity: {
+                            node('linux-small') {
+                                checkout scm
+                                script {
+                                    def packageFiles = findFiles(glob: '**/package.json')
+                                    for (int i = 0; i < packageFiles.size(); i++) {
+                                        dir(packageFiles[i].path.split('package.json')[0]) {
+                                            echo "Scanning ${packageFiles[i].name}"
+                                            nodejs(configId: 'npmrc-default', nodeJSInstallationName: 'nodejs') {
+                                                sh 'nsp check'
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                )
+            }
+        }
         stage('Deploy') {
             agent { label 'linux-small' }
             steps{
                 withMaven(maven: 'M3', globalMavenSettingsConfig: 'default-global-settings', mavenSettingsConfig: 'codice_settings.xml') {
                     checkout scm
                     sh 'mvn javadoc:aggregate -DskipStatic=true -DskipTests=true'
-                    sh 'mvn deploy -DskipStatic=true -DskipTests=true'
+                    retry(2) {
+                        sh 'mvn deploy -DskipStatic=true -DskipTests=true'
```

Comment:
We have avoided the multi-threaded deploys in the past because if something goes wrong it is hard to recover from. However since this is a snapshot it shouldn't be a problem. Adding the multi threaded option.